packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -1091,6 +1095,11 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
       );
       return;
     }
+    if (_codec == null) {
+      // codec was disposed during getNextFrame
+      return;
+    }
```

Comment:
The life cycle of the codec in this class is kinda unclear and it is hard to follow. For example, how can `getNextFrame` cause the disposal of the codec? Can we simplify the life cycle to make it easier to reason about to avoid future bugs?

Reply:
This behavior is tested in the “Decoding does not crash when disposed” test case. If a listener is removed during decoding and the listeners are empty, `_maybeDispose()` is executed and the codec is disposed.

https://github.com/flutter/flutter/blob/314870064a3a45e245be749cacb1375e643d0d91/packages/flutter/test/painting/image_stream_test.dart#L149

The codec lifecycle is confusing. Now that the codec is a mutable property, it needs to be checked after asynchronous work.